### PR TITLE
Enhance information for REST Hook triggers

### PR DESCRIPTION
**Pull request by armchairlinguist**
There was some useful information from the legacy docs that we didn't include in our REST Hook information in the current docs, especially around "reverse unsubscribe", so I've added that here.

I've also done some general cleanup on the Triggers page - making header levels consistent, adding links, wording changes for clarity and staying in sync with the product.

### CODE REVIEW

File: docs/_docs/triggers.md [hunk 1]
```diff
@@ -47,35 +47,35 @@ Once the settings are added, click _Save and Continue_ to add the new trigger an
...
 
-To add a polling trigger, select _Polling_ at the top of the settings page, then enter your API URL in the _API Endpoint_ field. If your API URL requires specific data in the URL, enter the following text in the URL where your API expects that data, replacing `key` for the input field key with the relevant data from the input form you created before:
+To add a polling trigger, select _Polling_ at the top of the settings page, then enter your API URL in the _API Endpoint_ field. If your API URL requires specific data in the URL path, enter the following text in the URL where your API expects that data, replacing `key` with the input field key containing the relevant data from the input form you created in the previous step:
 
 {% raw %}`{{bundle.inputData.key}}`{% endraw %}
 
-Otherwise, Zapier will automatically send any other input field data in the request body with the API call.
+Otherwise, Zapier will automatically include any input field data with the API call as URL parameters (for GET requests) or in the request body (for POST requests, e.g. for GraphQL APIs).
```
Comment (eliangcs) [inline docs/_docs/triggers.md:74]: I don't think we do GraphQL by default. Suggested change:

```suggestion
Otherwise, Zapier will automatically encode any input field data with the API call as URL parameters (for GET requests) or in the request body with JSON (for POST requests).
```

File: docs/_docs/triggers.md [hunk 2]
```diff
@@ -87,42 +87,58 @@ Alternately, if your app supports REST Hooks—webhook subscriptions that can be
...
 
 ![REST Hook Perform](https://cdn.zapier.com/storage/photos/5c6c154be13fbcad2a5dc8cb786f2dea.png)
 
-Finally, in the Perform, you can customize the code to evaluate the data your app's webhooks pass to Zapier. By default, Zapier includes `return [bundle.cleanedRequest];` to return the object from the webhook. If your data needs to be transformed, or includes multiple objects, add custom code to parse the response data in `bundle.cleanedRequest` within the Perform and turn it into an array of objects, such as [this example code](https://github.com/zapier/zapier-platform/blob/master/example-apps/rest-hooks/triggers/recipe.js#L42).
+Finally, in the Perform, you can customize the code to evaluate the data your app's webhooks pass to Zapier. By default, Zapier includes `return [bundle.cleanedRequest];` to return the request data from the webhook as an array. If your data needs to be transformed, or includes multiple objects, add custom code to parse the response data in `bundle.cleanedRequest` within the Perform and turn it into an array of objects, such as [this example code](https://github.com/zapier/zapier-platform/blob/master/example-apps/rest-hooks/triggers/recipe.js#L42).
```
Comment (eliangcs) [inline docs/_docs/triggers.md:115]: Let's pin the link to a specific commit in case the file is moved one day:

```suggestion
Finally, in the Perform, you can customize the code to evaluate the data your app's webhooks pass to Zapier. By default, Zapier includes `return [bundle.cleanedRequest];` to return the request data from the webhook as an array. If your data needs to be transformed, or includes multiple objects, add custom code to parse the response data in `bundle.cleanedRequest` within the Perform and turn it into an array of objects, such as [this example code](https://github.com/zapier/zapier-platform/blob/512f558ffa6dff11a0985c2e43c159d534bb6f36/example-apps/rest-hooks/triggers/recipe.js#L42).
```

File: docs/_docs/triggers.md [hunk 3]
```diff
@@ -87,42 +87,58 @@ Alternately, if your app supports REST Hooks—webhook subscriptions that can be
...
+
+Both the Perform and Perform List methods should return arrays, even if they only contain one object. The default Perform code adds an array around the cleaned webhook payload, so if your webhook already provides an array, you can remove this additional array and simply `return bundle.cleanedRequest;`.
```
Comment (eliangcs) [inline docs/_docs/triggers.md:117]: I'd rephrase:

```suggestion
Both the Perform and Perform List methods should return an array, even if they only contain one object. The default Perform code adds an array around the cleaned webhook payload, as in `return [bundle.cleanedRequest];`, so if your webhook already provides an array, you can simply `return bundle.cleanedRequest;`.
```

File: docs/_docs/triggers.md [hunk 4]
```diff
@@ -87,42 +87,58 @@ Alternately, if your app supports REST Hooks—webhook subscriptions that can be
...
+* Be mindful of Zapier's [rate limits](https://zapier.com/help/troubleshoot/behavior/rate-limits-and-throttling-in-zapier#step-4).
+* If your application receives a 410 response, that webhook subscription is no longer active, and you should stop sending data to it.
+* If your application receives repeated 4xx or 5xx failures from Zapier outside those error types, this can be handled at your discretion. You may choose to try again later, or to stop sending data and deactivate the hook.
 
-If for architectural reasons, your webhook will receive some data that shouldn't trigger the Zap, your code can return an empty array in those cases. If the Perform method returns an empty array, the Zap won't run.
+To signal that your app has deactivated the hook for any reason, you can optionally send a "reverse unsubscribe" call to Zapier. This can allow users to manage their subscriptions from inside your app, or permit you to clean up after a user deletes their account or revokes credentials. The request should be of the form `DELETE <target_url>`, where `target_url` is the unique target URL that was provided when the subscription was created. This will pause the Zap within Zapier.
```
Comment (eliangcs) [inline docs/_docs/triggers.md:133]: TIL! I wonder why it is called "reverse unsubscribe". Isn't "reverse unsubscribe" "subscribe"?

**Review by eliangcs — DISMISSED** [review]
Love the improvements! I have some suggested changes, but feel free to make your own judgement.

File: docs/_docs/triggers.md (hunk 1 above)
Comment (armchairlinguist) [reply docs/_docs/triggers.md:74]: Yeah, that's a good point. I used that as an example because GraphQL is the most familiar type of API for me that might use POSTs for polling, but it does give the wrong impression about the level of support.

I'm curious about why you mention `encode` - is that just because we'll also default to URL-encoding the params? From the point of view of the developer, we're including them in the request and the encoding is transparent, so I'm not sure about the best word to use.

File: docs/_docs/triggers.md (hunk 4 above)
Comment (armchairlinguist) [reply docs/_docs/triggers.md:133]: I think it's because it's an unsubscribe from the opposite direction 😄 But it is a bit funny in terms of terminology.

File: docs/_docs/triggers.md (hunk 3 above)
Comment (armchairlinguist) [reply docs/_docs/triggers.md:117]: You're right that arrays isn't clear here, but there are numeric agreement issues with `an array` with the rest of the sentence, and the default code is mentioned in the previous paragraph, so I may make a slightly different adjustment on this.

**Comment by armchairlinguist** [comment]
Thanks for the thoughtful feedback, @eliangcs ! Mind taking another look? I adopted some of your suggestions, and some changes that I hope have similar effect.

**Review by eliangcs — APPROVED** [review]
Looks good, thank you!
